pkg/controllers/management/drivers/kontainerdriver/kontainerdriver.go

```diff
@@ -450,3 +461,12 @@ func (l *Lifecycle) removeFieldFromCluster(obj *v3.KontainerDriver) error {
 
 	return nil
 }
+
+func RemoveCondition(obj *v3.KontainerDriver, cond condition.Cond) {
+	for i, c := range obj.Status.Conditions {
```

Comment:
is it possible to utilize our condition framework? Also I'd rather use condition `Downloaded` that can be set to true/false/unknown, where unknown is transitioning. And it can be set to true only once (we have a framework method for that)

If we use Downloading condition instead, it can result in framework inconsistencies. One I can think of is condition-based state calculation being broken, when we treat True as active

Reply:
I am using `Downloaded` to do what you are suggesting. If the binary is missing, set `Downloaded` to `Unknown`. If the download process success, set `Downloaded` to `True`.  Set it to false if download fails.

But I am not sure about that we should remove conditions or not. I am removing the `Inactive` condition when I activate a driver.